bool enable
string mode
---
bool success
